Assets/HexMap.cs

```diff
@@ -215,7 +188,9 @@ public Hex GetAcceptableStartPosition()
             int impassibleTiles = 0;
             foreach (Hex h in neighbors)
             {
-                if (h.GetTerrain() == Terrain.MOUNTAIN || h.GetTerrain() == Terrain.WATER)
+                if (h == null)
+                    Debug.LogWarning("Null neighbor, something is borked");
+                else if (h.GetTerrain() == Terrain.MOUNTAIN || h.GetTerrain() == Terrain.WATER)
```

Comment:
Abstracted method that you pass in the terrain and it tells returns a type, then use case statement.